application/src/main/java/org/thingsboard/server/actors/ruleChain/RuleNodeActorMessageProcessor.java

```diff
@@ -89,6 +89,7 @@ public void onUpdate(TbActorCtx context) throws Exception {
     public void stop(TbActorCtx context) {
         if (tbNode != null) {
             tbNode.destroy();
+            tbNode.onRemove(defaultCtx);
```

Comment:
This breaks backward-compatibility. You should provide a default implementation of the "onRemove" that calls "destroy" in the TbNode interface.

Reply:
I am confused. Why do you need a separate method if you use it only in pair with ```tbNode.destroy();```